### PR TITLE
`align*` clashes with `split` and `equation*`

### PR DESCRIPTION
**Pull request by likanzhan**
`align*` clashes with `split` and `equation*`

### CODE REVIEW

**Comment by mortenpi** [comment]
I assume you mean that the alignment is wrong? It's interesting that `align` within `equation* / split` does work.  I can repro that with this: https://gist.github.com/mortenpi/3ef142a957a015ade5a69a77fc8acdcf

align

The patch LGTM, but another set of eyes wouldn't hurt. @fredrikekre?

**Comment by likanzhan** [comment]
Thanks for the example and pointing out the alignment problem.
But actually, the crucial error occurs when the enclosed text includes multiple `&`s, such as

1. Not working (Error occurs): equation* / split:

```
\begin{equation*}
\begin{split}
 x^2 && x^3 && x^4\\
\end{split}
\end{equation*}
```

2. Not working (Error occurs): equation* / split / align*:

```
\begin{equation*}
\begin{split}
\begin{align*}
 x^2 && x^3 && x^4\\
\end{align*}
\end{split}
\end{equation*}
```

3. Working: align*:

```
\begin{align*}
    x^2 && x^3 && x^4\\
\end{align*}
```

File: src/Writers/LaTeXWriter.jl [hunk 1]
```diff
@@ -559,9 +559,13 @@ end
 # This (equation*, split) math env seems to be the only way to correctly
 # render all the equations in the Julia manual.
 function latex(io::IO, math::Markdown.LaTeX)
-    _print(io, "\\begin{equation*}\n\\begin{split}")
-    _print(io, math.formula)
-    _println(io, "\\end{split}\\end{equation*}")
+    if occursin(r"^\\begin\{align\*\}", math.formula)
```
[the remark this answers is not in the document]
Comment (likanzhan) [reply src/Writers/LaTeXWriter.jl:562]: Yes, That's right.

**Comment by mortenpi** [comment]
Thanks for the contribution @likanzhan!

**Pull request merged**